Cython/Utility/ModuleSetupCode.c

```diff
@@ -351,15 +284,18 @@
     #define CYTHON_USE_ASYNC_SLOTS 1
   #endif
   #ifndef CYTHON_USE_PYLONG_INTERNALS
-    #define CYTHON_USE_PYLONG_INTERNALS 1
+    #define CYTHON_USE_PYLONG_INTERNALS (!CYTHON_COMPILING_IN_CPYTHON_NOGIL)
   #endif
-  #ifndef CYTHON_USE_PYLIST_INTERNALS
+  #ifdef Py_GIL_DISABLED
```

Comment:
We should have one place where we decide about internal configuration macros, and then use only them.
```suggestion
  #if CYTHON_COMPILING_IN_CPYTHON_NOGIL
```